include/clink/feature/one_hot_encoder.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ONE_HOT_ENCODER_H_
+#define ONE_HOT_ENCODER_H_
+
+#include "clink/feature/proto/one_hot_encoder.pb.h"
+#include "clink/linalg/sparse_vector.h"
+
+namespace clink {
+
+// Params of OneHotEncoderModel.
+typedef struct OneHotEncoderParams {
+  // Whether to drop the last category.
+  bool isDropLast;
+} OneHotEncoderParams;
+
+// A Model which encodes data into one-hot format.
+class OneHotEncoderModel {
```

Comment:
Do we need to define `class Model` in C++ similar to what we did in Flink ML? If so, we can do this either in this PR or in a followup PR.

Reply:
I think we can define the infra classes in a followup PR, as it can introduce discussions not directly related to OneHotEncoder's implementations.